### ISSUE TITLE
Allow configuration cache location

### ISSUE THREAD

**Issue by ericcornelissen** [plantain-00/type-coverage#116]
#### Expected:

An option to configure the location of the type-coverage cache.

It's really just a quality of life feature request, I installed type-coverage on one of my project and would like to put it's cache in `.cache/type-coverage/` instead of `.type-coverage/`

#### Actual:

There's only an option to enable or disable the cache.

**Comment by plantain-00** [comment]
`v2.24.0` supports this by `--cache-directory`

**Comment by ericcornelissen** [comment]
Thanks @plantain-00 :rocket: 

I gave it a try and it seems to work quite well. One piece of feedback: in the case of a nested target directory, type-coverage fails if the parent folder does not yet exist. I think ideally the entire directory structure would be created if necessary.

Example:

```sh
$ rm -rf .cache
$ npx type-coverage --cache --cache-directory .cache/type-coverage
ENOENT: no such file or directory, mkdir '.cache/type-coverage'
```

For the record, I tried this on _Ubuntu 22.04.1 LTS_, _Node v18.12.1_, and _npm v8.19.2_.
